hoomd/md/methods/methods.py

```diff
@@ -133,265 +75,250 @@ class ConstantVolume(Thermostatted):
-            :math:`[\mathrm{time}]`.
-
-        translational_thermostat_dof (tuple[float, float]): Additional degrees
-            of freedom for the translational thermostat (:math:`\xi`,
-            :math:`\eta`)
+        thermostat (hoomd.md.methods.thermostats.Thermostat): Temperature
+            control for the integrator
 
-        rotational_thermostat_dof (tuple[float, float]): Additional degrees
-            of freedom for the rotational thermostat (:math:`\xi_\mathrm{rot}`,
-            :math:`\eta_\mathrm{rot}`)
+    .. _Kamberaj 2005: http://dx.doi.org/10.1063/1.1906216
     """
 
     def __init__(self, filter, thermostat=None):
-
+        super().__init__()
         # store metadata
         param_dict = ParameterDict(filter=ParticleFilter,
-                                   kT=Variant,
-                                   thermostat=Thermostat,
-                                   translational_thermostat_dof=(float, float),
-                                   rotational_thermostat_dof=(float, float))
-        if thermostat is None:
-            thermostat = ConstantEnergy()
-        param_dict.update(
-            dict(filter=filter,
-                 translational_thermostat_dof=(0, 0),
-                 rotational_thermostat_dof=(0, 0),
-                 thermostat=thermostat))
+                                   thermostat=OnlyTypes(Thermostat,
+                                                        allow_none=True))
+        param_dict.update(dict(filter=filter, thermostat=thermostat))
         # set defaults
         self._param_dict.update(param_dict)
 
     def _attach_hook(self):
-        if self.thermostat._attached and self.thermostat._simulation != self._simulation:
-            self.thermostat = copy.deepcopy(self.thermostat)
-        self.thermostat._attach(self._simulation)
-        if self.thermostat._cpp_obj.requiresSeed():
-            self._simulation._warn_if_seed_unset()
         # initialize the reflected cpp class
         if isinstance(self._simulation.device, hoomd.device.CPU):
-            my_class = _md.TwoStepConstantVolume
+            cls = _md.TwoStepConstantVolume
             thermo_cls = _md.ComputeThermo
         else:
-            my_class = _md.TwoStepConstantVolumeGPU
+            cls = _md.TwoStepConstantVolumeGPU
             thermo_cls = _md.ComputeThermoGPU
 
         group = self._simulation.state._get_group(self.filter)
         cpp_sys_def = self._simulation.state._cpp_sys_def
-        thermo = thermo_cls(cpp_sys_def, group)
-        self._cpp_obj = my_class(cpp_sys_def, group, thermo, self.thermostat._cpp_obj)
-        super()._attach_hook()
-
+        self._thermo = thermo_cls(cpp_sys_def, group)
 
+        if self.thermostat is None:
+            self._cpp_obj = cls(cpp_sys_def, group, None)
+        else:
+            if self.thermostat._attached:
+                raise RuntimeError("Trying to attach a thermostat that is "
+                                   "already attached")
```

Comment:
This shouldn't be needed since if you set an attached thermostat, the code errors.